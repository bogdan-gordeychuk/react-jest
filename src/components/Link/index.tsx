interface ILinkProps {
  href: string;
  className?: string;
  children: React.ReactNode | React.ReactNode[];
}

function Link({ children, href, className }: ILinkProps) {
  return (
    <a href={href} className={className}>
      {children}
    </a>
  );
}

export default Link;
