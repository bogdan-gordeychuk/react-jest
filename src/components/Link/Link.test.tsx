import { render } from '@testing-library/react';
import Link from '.';

test('renders link', () => {
  const { asFragment } = render(<Link className="mylink" href="/some-page">Hello!</Link>);
  expect(asFragment()).toMatchSnapshot();
});
