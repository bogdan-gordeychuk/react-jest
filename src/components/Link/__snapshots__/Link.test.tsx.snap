// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders link 1`] = `
<DocumentFragment>
  <a
    class="mylink"
    href="/some-page"
  >
    Hello!
  </a>
</DocumentFragment>
`;
